# **Caso de estudio: Detección de riesgo de Alzheimer**
En este documento se desarrolla y analiza un modelo Naive Bayes para la identificación
temprana de pacientes con diagnóstico de Alzheimer, basándose en variables clínicas,
demográficas y de estilo de vida del paciente.

De acuerdo con lo anterior, las variables utilizadas son todas las variables numericas del dataset para mayor precisión del modelo:

**Age**: Edad del paciente en años al momento del estudio.

**Gender**: Género del paciente (0 = Femenino, 1 = Masculino).

**Ethnicity**: Etnia del paciente (0 = Caucásico, 1 = Afroamericano, 2 = Asiático, 3 = Otro).

**EducationLevel**: Nivel educativo del paciente (0 = Ninguno, 1 = Secundaria, 2 = Universitario, 3 = Posgrado).

**BMI**: Índice de masa corporal del paciente.

**Smoking**: Indica si el paciente fuma (0 = No, 1 = Sí).

**AlcoholConsumption**: Consumo semanal de alcohol en unidades.

**PhysicalActivity**: Actividad física semanal en horas.

**DietQuality**: Puntuación de la calidad de la dieta del paciente (0-10).

**SleepQuality**: Puntuación de la calidad del sueño del paciente (4-10).

**FamilyHistoryAlzheimers**: Indica si hay antecedentes familiares de Alzheimer (0 = No, 1 = Sí).

**CardiovascularDisease**: Indica si el paciente tiene enfermedad cardiovascular (0 = No, 1 = Sí).

**Diabetes**: Indica si el paciente tiene diabetes (0 = No, 1 = Sí).

**Depression**: Indica si el paciente tiene depresión (0 = No, 1 = Sí).

**HeadInjury**: Indica si el paciente ha sufrido una lesión en la cabeza (0 = No, 1 = Sí).

**Hypertension**: Indica si el paciente tiene hipertensión (0 = No, 1 = Sí).

**SystolicBP**: Presión arterial sistólica del paciente (mmHg).

**DiastolicBP**: Presión arterial diastólica del paciente (mmHg).

**CholesterolTotal**: Nivel de colesterol total del paciente (mg/dL).

**CholesterolLDL**: Nivel de colesterol LDL del paciente (mg/dL).

**CholesterolHDL**: Nivel de colesterol HDL del paciente (mg/dL).

**CholesterolTriglycerides**: Nivel de triglicéridos del paciente (mg/dL).

**MMSE**: Puntuación del Mini-Mental State Examination. Evalúa la función cognitiva del paciente (0-30).

**FunctionalAssessment**: Evaluación de la capacidad funcional del paciente (0-10).

**MemoryComplaints**: Indica si el paciente reporta quejas de memoria (0 = No, 1 = Sí).

**BehavioralProblems**: Indica si el paciente presenta problemas de comportamiento (0 = No, 1 = Sí).

**ADL**: Puntuación de actividades de la vida diaria del paciente (0-10).

**Confusion**: Indica si el paciente presenta episodios de confusión (0 = No, 1 = Sí).

**Disorientation**: Indica si el paciente presenta desorientación (0 = No, 1 = Sí).

**PersonalityChanges**: Indica si el paciente presenta cambios de personalidad (0 = No, 1 = Sí).

**DifficultyCompletingTasks**: Indica si el paciente tiene dificultad para completar tareas (0 = No, 1 = Sí).

**Forgetfulness**: Indica si el paciente presenta olvidos frecuentes (0 = No, 1 = Sí).

0. Se procede con la carga de las librerías de trabajo

In [1]:
import numpy as np
import pandas as pd

from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import confusion_matrix

1. Se cargan los datos de trabajo de la base de datos Altzheimer disease

In [2]:
nxl='/content/Alzheimer_disease_dataset.xlsx'
XDB=pd.read_excel(nxl,sheet_name=0)
XDB.head(100)

XD = XDB[['Age', 'Gender', 'Ethnicity', 'EducationLevel', 'BMI', 'Smoking',
           'AlcoholConsumption', 'PhysicalActivity', 'DietQuality', 'SleepQuality',
           'FamilyHistoryAlzheimers', 'CardiovascularDisease', 'Diabetes', 'Depression',
           'HeadInjury', 'Hypertension', 'SystolicBP', 'DiastolicBP', 'CholesterolTotal',
           'CholesterolLDL', 'CholesterolHDL', 'CholesterolTriglycerides', 'MMSE',
           'FunctionalAssessment', 'MemoryComplaints', 'BehavioralProblems', 'ADL',
           'Confusion', 'Disorientation', 'PersonalityChanges',
           'DifficultyCompletingTasks', 'Forgetfulness']]
XD.head(10)

yd = XDB['Diagnosis']
yd.head()

,Diagnosis
0,0
1,0
2,0
3,0
4,0


2. Implementamos modelo Naive Bayes

In [3]:
np.set_printoptions(precision=3)
mnb = GaussianNB()
mnb.fit(XD, yd)

u     = mnb.theta_
sigma = mnb.var_; sigma = np.sqrt(sigma)

print("'Age','Gender','Ethnicity','EducationLevel','BMI','Smoking','AlcoholConsumption',")
print("'PhysicalActivity','DietQuality','SleepQuality','FamilyHistoryAlzheimers',")
print("'CardiovascularDisease','Diabetes','Depression','HeadInjury','Hypertension',")
print("'SystolicBP','DiastolicBP','CholesterolTotal','CholesterolLDL','CholesterolHDL',")
print("'CholesterolTriglycerides','MMSE','FunctionalAssessment','MemoryComplaints',")
print("'BehavioralProblems','ADL','Confusion','Disorientation','PersonalityChanges',")
print("'DifficultyCompletingTasks','Forgetfulness'")
print()
print("Medias por categoría (fila 0 = Sin Alzheimer, fila 1 = Con Alzheimer):")
print(u)
print("Las desviaciones son:")
print(sigma)
print()
print("Los límites superiores de las variables son: ", u + sigma)
print()
print("Los límites inferiores de las variables son: ", u - sigma)

'Age','Gender','Ethnicity','EducationLevel','BMI','Smoking','AlcoholConsumption',
'PhysicalActivity','DietQuality','SleepQuality','FamilyHistoryAlzheimers',
'CardiovascularDisease','Diabetes','Depression','HeadInjury','Hypertension',
'SystolicBP','DiastolicBP','CholesterolTotal','CholesterolLDL','CholesterolHDL',
'CholesterolTriglycerides','MMSE','FunctionalAssessment','MemoryComplaints',
'BehavioralProblems','ADL','Confusion','Disorientation','PersonalityChanges',
'DifficultyCompletingTasks','Forgetfulness'

Medias por categoría (fila 0 = Sin Alzheimer, fila 1 = Con Alzheimer):
[[7.495e+01 5.140e-01 7.084e-01 1.316e+00 2.752e+01 2.901e-01 1.007e+01
  4.908e+00 4.975e+00 7.125e+00 2.628e-01 1.361e-01 1.591e-01 2.023e-01
  9.719e-02 1.397e-01 1.346e+02 8.978e+01 2.250e+02 1.254e+02 5.873e+01
  2.266e+02 1.627e+01 5.861e+00 1.159e-01 9.647e-02 5.708e+00 2.109e-01
  1.649e-01 1.562e-01 1.562e-01 3.017e-01]
 [7.484e+01 4.921e-01 6.776e-01 1.233e+00 2.791e+01 2.855e-01 9.980e+00
  4.943e+00

**Análisis de Resultados**

De los resultados arrojados por el modelo, podemos observar que las variables con
mayores diferencias entre la categoría sin Alzheimer y con Alzheimer son
FunctionalAssessment, ADL, MMSE, MemoryComplaints y BehavioralProblems, lo que
confirma que estas son las variables más determinantes para la clasificación.

La variable FunctionalAssessment presenta una media de 5.86 en pacientes sin Alzheimer
frente a 3.65 en pacientes con Alzheimer, lo que indica que una menor capacidad
funcional está fuertemente asociada al diagnóstico positivo. De manera similar, la
variable ADL muestra una media de 5.71 en el grupo sin Alzheimer frente a 3.66 en el
grupo con Alzheimer, evidenciando que los pacientes diagnosticados presentan una
mayor dificultad para realizar actividades de la vida diaria.

La variable MMSE, que evalúa la función cognitiva del paciente, registra una media de
16.27 en el grupo sin Alzheimer y de 11.99 en el grupo con Alzheimer, reflejando un
deterioro cognitivo significativamente mayor en los pacientes diagnosticados.

Las variables MemoryComplaints y BehavioralProblems también presentan diferencias
notables entre categorías. MemoryComplaints registra una media de 0.116 en el grupo
sin Alzheimer frente a 0.376 en el grupo con Alzheimer, mientras que BehavioralProblems
muestra 0.096 versus 0.267 respectivamente, lo que indica que los pacientes
diagnosticados reportan con mayor frecuencia quejas de memoria y problemas de
comportamiento.

Las demás variables como Age, Gender, BMI, SystolicBP, DiastolicBP, CholesterolTotal,
CholesterolLDL, CholesterolHDL, Smoking, AlcoholConsumption, PhysicalActivity y
DietQuality no presentan diferencias significativas entre las categorías, lo que indica
que estas características por sí solas no son determinantes para la clasificación del
diagnóstico de Alzheimer.


3. Se procede con la evaluación del modelo. Para la evaluación de este tipo de modelo se utiliza la matriz de confusión.

In [4]:
ydp=mnb.predict(XD) #Esto es lo que el modelo aprende - ydp de pronóstico
cm=confusion_matrix(yd,ydp)
print(cm)

VN=cm[0,0]; FP=cm[0,1]; FN=cm[1,0]; VP=cm[1,1]; TDatos=len(XDB)

#1. Exactitud: Funcionamiento general del modelo (Primera métrica, como funcionó pronosticando)
Ex=(VP+VN)/(TDatos)
print("Exactitud: ",Ex)

#2. Tasa de error: % fallos del sistema
TEr=(FP+FN)/(TDatos)
print("Tasa de error: ",TEr) #Modelo que pase del 75% es muy bueno

#3. Sensibilidad: Como se  comportó con respecto a los positivos. (fila)
Se=VP/(VP+FN)
print("Sensibilidad: ",Se)

#4. Especificidad: Como se comporta pronosticando negativos
Es=VN/(VN+FP)
print("Especificidad: ",Es)

#5. Precisión: Es una versión de como se comporta el modelo frente a los positivos solamente. (columna)
Pr=VP/(VP+FP)
print("Precisión: ",Pr)

#6. Predicción Negativa: como funciona el modelo pronosticando créditos PreNegados
PrN=VN/(VN+FN)
print("Predicción Negativa: ",PrN)

[[1205  184]
 [ 217  543]]
Exactitud:  0.8134015821312238
Tasa de error:  0.18659841786877618
Sensibilidad:  0.7144736842105263
Especificidad:  0.8675305975521959
Precisión:  0.7469050894085282
Predicción Negativa:  0.8473980309423348


**Análisis de Resultados**

De acuerdo con los resultados obtenidos, se puede observar de manera general que el
modelo alcanzó una exactitud del 81.3%, lo que indica un buen desempeño en la
clasificación de los pacientes entre las categorías con y sin diagnóstico de Alzheimer.

Se destaca la especificidad del 86.8%, lo que indica que el modelo reconoce
correctamente a la gran mayoría de los pacientes que realmente no presentan diagnóstico
de Alzheimer. De igual manera, la predicción negativa del 84.7% confirma que la mayoría
de los pacientes clasificados como sin Alzheimer realmente no lo tienen.

La sensibilidad del 71.4% indica que el modelo logra identificar correctamente a un
porcentaje significativo de los pacientes que sí presentan diagnóstico de Alzheimer,
detectando 543 de los 760 casos positivos de la base de datos. La precisión del 74.7%
evidencia que cuando el modelo clasifica a un paciente como con Alzheimer, en la gran
mayoría de los casos el diagnóstico es correcto.

En comparación con el modelo aplicado al dataset de Stroke, este modelo presenta un
desempeño notablemente superior en todas las métricas, especialmente en sensibilidad,
lo que se explica por el mejor balance entre categorías que presenta este dataset
(64.6% sin Alzheimer vs 35.4% con Alzheimer).


4. Se evalúa a un paciente XP= [75, 1, 0, 1, 28.5, 1, 5.0, 2.0, 4.0, 5.0, 1, 1, 1, 0, 0, 1, 145, 90, 220, 140, 45, 180, 18, 4.0, 1, 1, 4.0, 1, 1, 1, 1, 1]

In [5]:
XP = [75, 1, 0, 1, 28.5, 1, 5.0, 2.0, 4.0, 5.0,
      1, 1, 1, 0, 0, 1, 145, 90, 220, 140, 45, 180,
      18, 4.0, 1, 1, 4.0, 1, 1, 1, 1, 1]
ydc = mnb.predict([XP])
print(ydc)

if ydc == 1:
    print("El paciente tiene diagnóstico positivo de Alzheimer")
else:
    print("El paciente no presenta diagnóstico de Alzheimer")

[1]
El paciente tiene diagnóstico positivo de Alzheimer


/usr/local/lib/python3.12/dist-packages/sklearn/utils/validation.py:2739: UserWarning: X does not have valid feature names, but GaussianNB was fitted with feature names
  warnings.warn(
